Update how default browser config values are assigned

B77Mills opened the pull request:
This update will ensure that all browser config values have a their default values applied.

I also removed the withGTM prop as it was no longer referenced and config was also not passed through anywhere.

B77Mills committed:
Ensure brwoser config defaults are applied correctly

Instead of just taking config as a prop spread the default config and config overrides bein sent in.

B77Mills committed:
Remove unused prop

B77Mills committed:
Clean prop assinment

solocommand reviewed (changes requested):
@B77Mills I think this is fine, assuming that we _want_ the defaults to be applied to everyone. That said, `idxArgs` and `inquiryArgs` still need to have a default/fallback to an empty object (in case someone sends `Monorail(Browser, { idxArgs: null })`).

Update lines 89, 91, and 94 to use `idxArgs || {}` and `inquiryArgs || {}`.

B77Mills committed:
Use getAsAObject to set idxArgs & inquiryArgs

This will ensure these parts of the config being passed in are at minimum an empty object.

B77Mills commented:
@solocommand  I updated the setting of those values to use getAs object to ensure they are, at minimum and empty, object.
